R/plot.betadisper.R

```diff
@@ -1,10 +1,26 @@
 `plot.betadisper` <- function(x, axes = c(1,2), cex = 0.7, pch = seq_len(ng),
                               col = NULL, hull = TRUE,
+                              ellipse = FALSE, ellipse.type = c("sd","se"),
+                              ellipse.conf = NULL,
                               ylab, xlab, main, sub, ...)
 {
     localAxis <- function(..., col, bg, pch, cex, lty, lwd) axis(...)
     localBox <- function(..., col, bg, pch, cex, lty, lwd) box(...)
     localTitle <- function(..., col, bg, pch, cex, lty, lwd) title(...)
+    Ellipse <- function(scrs, type, conf, col, lty, lwd) {
+        mat <- cov.wt(scrs, wt = rep_len(1, nrow(scrs)))
```

Comment:
This can be simplified slightly: weights are not needed if you have unweighted covariance, but you can write
```R
mat <- cov.wt(scrs)
```
and then change line 15 below into
```R
if (type == "se") {
    mat$cov <- mat$cov / mat$n.obs
}
```
Using `cov.wt` is practical, though, because it gives you both the covariances and the centroids.